bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/WebSocketBase.java

```diff
@@ -89,10 +93,17 @@ public void onWebSocketConnect(@Nullable Session session) {
 
     void close() {
         logger.debug("{} connection close requested", this.getClass().getSimpleName());
+
         Session session = getSession();
         if (session != null) {
             session.close();
         }
+
+        final Future<?> sessionFuture = this.sessionFuture;
+        logger.trace("Closing session Future: {}", sessionFuture);
+        if (!sessionFuture.isDone()) {
```

Comment:
No need to check that sessionFuture is not null ?

Reply:
Good point. I think this is merely for the codestyle as close is never called before connect, but yes.